pkg/cloud/services/ec2/instances.go

```diff
@@ -1307,3 +1318,34 @@ func getInstanceCPUOptionsRequest(cpuOptions infrav1.CPUOptions) *types.CpuOptio
 
 	return request
 }
+
+func (s *Service) shouldEnablePrimaryIpv6(i *infrav1.Instance) (bool, error) {
```

Comment:
This method is making the assumption that when IPv6 is enabled, IPv6 would be the primary. If correct, could you please add that as a comment to this method?

Reply:
Right, `PrimaryIpv6` attribute will be enabled on the instances when `spec.network.vpc.ipv6` is non-nil (i.e. user intention to use IPv6) and the subnet to launch the instance has IPv6 (i.e. IPv6-only or dualstack).

I will add the above comment in. Thanks!